syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.search.image.protoc";
option java_outer_classname = "ImageSearchProto";

// The greeting service definition.
service Greeter {
  rpc Classify (ImgRequest) returns (ExecuteReply) {};
  rpc Search (ImgRequest) returns (ExecuteReply) {};
}

message ImgRequest {
  string content = 1;
  string book_name = 2;
  int32 previous_page_num = 3;
  string user_id = 4;
}

// The response message containing the greetings
message ExecuteReply {
  string book_name = 1;
  int32 page_num = 2;
  int32 code = 3;
  string message = 4;
  string executed_time = 5;
  string user_id = 6;
  bool is_last_page = 7;
}